CenterForOpenScience/ember-osf-web
# [EMB-341] Tokens page wrap-up

## Purpose
Get the tokens page ready to waffle.


## Summary of Changes
- Add `delete-button` component, for all our delete button needs
- Respond to demo feedback: Update tokens page layout, language; improve loading states; validate token name length
- Add tokens page acceptance tests
- More thoroughly use types throughout our mirage code

Demo:
![tokens](https://user-images.githubusercontent.com/6776190/45493898-de951380-b73d-11e8-912e-a6280024d6a0.gif)



## Side Effects
n/a


## Feature Flags
`ember_user_settings_tokens_page`: enable the tokens page


## QA Notes
- Go to `/settings/tokens`
- Create, edit, delete all sorts of tokens
- The layout shouldn't jump around when moving through pages
- Check that "This is the only time your token will be displayed." isn't a lie


## Ticket


https://openscience.atlassian.net/browse/EMB-341

# Reviewer Checklist
